src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -336,6 +338,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     internal var superclass: TypeName = ANY
     internal val superclassConstructorParameters = mutableListOf<CodeBlock>()
     internal val superinterfaces = mutableListOf<TypeName>()
+    internal val superinterfaceDelegates = mutableListOf<ParameterSpec>()
```

Comment:
There needs to be at least something which keeps track of them. What if someone reverts the typespec to a builder, iterates over all of the functions' parameters, and changes the name of each parameter? I can't think of any other way except direct reference at the moment